File: C# Activities/TestApp/Array2Activitie1.cs
using System;

class Array2Activitie1
{
    static void Main()
    {
        Console.Write("¿De qué tamaño será la matriz?: ");
        int n = int.Parse(Console.ReadLine());

        int[] valores = new int[n];

        // Pedir los valores
        for (int i = 0; i < n; i++)
        {
            Console.Write($"Ingrese el valor para la posición {i + 1}: ");
            valores[i] = int.Parse(Console.ReadLine());
        }

        // Mostrar los valores ingresados
        Console.WriteLine("\nLos valores ingresados son:");
        foreach (int v in valores)
        {
            Console.Write(v + " ");
        }
        Console.WriteLine();

        // Pedir posición y valor a insertar
        Console.Write($"\nIngrese la posición donde desea insertar (0 a {n}): ");
 int pos = int.Parse(Console.ReadLine()!);



        Console.Write("Ingrese el valor a insertar: ");
        int valor = int.Parse(Console.ReadLine()!);

        // Crear un arreglo más grande para insertar
        int[] nuevaLista = new int[n + 1];

        // Copiar y desplazar elementos
        for (int i = 0; i < pos; i++)
        {
            nuevaLista[i] = valores[i];
        }

        nuevaLista[pos] = valor;

        for (int i = pos; i < n; i++)
        {
            nuevaLista[i + 1] = valores[i];
        }

        // Mostrar la nueva lista
        Console.WriteLine("\nLa nueva lista es:");
        foreach (int v in nuevaLista)
        {
            Console.Write(v + " ");
        }
        Console.WriteLine();
    }
}

File: Activities Python/ActivitiesArray/Array2 Activitie1.py
n = int(input(f"De que tamaño sera la matriz?: ")) # se le pregunta el usuario que tan grande sera el arreglo
valores = [0] * n #es porque se inicializa en 0
for i in range(n):
    valores[i] = int(input(f"Ingrese el valor para la posicion {i+1}: ")) #La informacion se guarda en la lista i

print(f"Los valores ingresados son: {valores}")
pos = int(input(f"\nIngrese la posicion donde desea insertar (0 a {n}): "))
valor = int(input("Ingrese el valor a insertar: "))

valores = valores + [0] #Se suma un nuevo espacio al final de la lista

for i in range(len(valores)-1, pos, -1): #Se desplazan los elementos a la derecha, el primer -1 indica el último índice, el pos indica la posicion antes que parara y el ultimo -1 indica la direccion
    valores[i] = valores[i-1]

valores[pos] = valor

print("\nLa nueva Lista es:")
print(valores)
